pipeline_steps/run_tests.groovy

```diff
@@ -0,0 +1,38 @@
+def run_irr_tests() {
+  common.conditionalStage(
+    stage_name: "Run IRR Tests",
+    stage: {
+      pubcloud.runonpubcloud {
+        try {
+          ansiColor('xterm') {
+            dir(env.WORKSPACE) {
+              print("Triggered by PR: ${env.ghprbPullLink}")
+              checkout([$class: 'GitSCM',
+                branches: [[name: "origin/pr/${env.ghprbPullId}/merge"]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [[$class: 'CleanCheckout']],
+                submoduleCfg: [],
+                userRemoteConfigs: [
+                  [
+                    url: "https://github.com/${env.ghprbGhRepository}.git",
+                    refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
+                  ]
+                ]
+              ]) // checkout
+              sh """#!/bin/bash
+              pushd \$(basename ${env.ghprbGhRepository})
+                bash ./run_tests.sh
```

Comment:
The form `bash script.sh` will ignore opts set in the script's `#!` line, but if the script sets options via the `set` command then this will be fine.